packages/dapp/src/const/MenuConfig/useWalletSelectContent.tsx

```diff
@@ -34,43 +50,62 @@ export const useWalletSelectContent = () => {
     [connect, onWalletConnect],
   );
 
+  useEffect(() => {
+    verifyMultisigEnvironment();
+  }, [account?.address]);
+
   const walletMenuItems = useMemo<MenuListItem[]>(() => {
-    const installedWallets = supportedWallets.filter((wallet) =>
-      wallet.installed(),
+    const installedWallets = supportedWallets.filter(
+      async (wallet) => await wallet.installed(),
     );
 
     const notInstalledWallets = supportedWallets.filter(
       (wallet) => !wallet.installed() && wallet.name !== 'Default Wallet', // always remove Default Wallet from not installed Wallets
     );
```

Comment:
We should also adjust this part to utilize the async `installed` method and any other place if any.